### ISSUE TITLE
Update composer.json to remove .git files

### ISSUE THREAD

**Issue by johnpicozzi** [oomphinc/drupal-scaffold#22]
Recently on current project we found that we had to add the following to remove .git files to allow for deploying dev modules. 

```
    "scripts": {
        "remove-git-submodules": "find . -mindepth 2 -type d -name .git | xargs rm -rf",
        "post-install-cmd": [
            "DrupalComposer\\DrupalScaffold\\Plugin::scaffold",
            "@remove-git-submodules"
        ],
        "post-update-cmd": [
            "DrupalComposer\\DrupalScaffold\\Plugin::scaffold",
            "@remove-git-submodules"
        ],
        "test:php": [
            "vendor/bin/phpunit"
        ],
        "validate:php": [
            "vendor/bin/phpcs"
        ]
    }, 
```

More info can be found here: https://pantheon.io/docs/guides/partial-composer/

**Comment by nathandentzau** [comment]
This _could_ be dangerous for any projects with actual submodules. I'm thinking a composer plugin would work for this. I'll see what I can come up with.

**Comment by balbuf** [comment]
I don't see why this should be the job of composer - shouldn't `.git` just be excluded from whatever mechanism drives the deploy (e.g. rsync exclude or gitignore)? At the very least, if there is no other way, the `.git` directories should only be deleted in the deploy workspace and not locally for any user.

Besides, the terminology is misleading; while this would succeed in deleting the `.git` directories of real git submodules (and would likely wreak havoc on a system that's already clunky - thankfully we don't really use git submodules), composer packages are _not_ installed as git submodules.

Further, for people that wish to make patches for any packages they are using, having those packages installed via their source (i.e. with the `.git` directory) is incredibly helpful.

In short, I think this is a bad idea.

**Comment by johnpicozzi** [comment]
After reviewing your comments and talking with @marcaddeo I removed the `composer.json` addition and updated the `.rsyn-filter` file to include .git directories. I will post a patch this weekend to add to the file and update the comment (it's confusing).

**Comment by nathandentzau** [comment]
Closing this for now. If we have a solution feel free to open a new issue.
